integration-test/test.sh

```diff
@@ -0,0 +1,350 @@
+#!/bin/bash
+set -eou pipefail
+IFS=$'\n\t'
+
+# these flighting values should match Makefile
+readonly TEST_SUBS="c3dfd792-49a4-4b06-80fc-6fc6d06c4742"
+readonly TEST_REGION="Brazil South"
+
+# supported images (add/update them as new major versions come out)
+readonly DISTROS=(
+	"2b171e93f07c4903bcad35bda10acf22__CoreOS-Beta-877.1.0" \
+	"5112500ae3b842c8b9c604889f8753c3__OpenLogic-CentOS-71-20150731" \
+	"b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-15_04-amd64-server-20151201-en-us-30GB" \
+	"b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_3-LTS-amd64-server-20151117-en-us-30GB" )
+
+# Test constants
+readonly SCRIPT_DIR=$(dirname $0)
+readonly CONCURRENCY=10
+readonly DOCKER_CERTS_DIR=dockercerts
+readonly VM_PREFIX=dockerexttest-
+readonly VM_USER=azureuser
+readonly EXTENSION_NAME=DockerExtension
+readonly EXTENSION_PUBLISHER=Microsoft.Azure.Extensions
+readonly EXTENSION_VERSION=1.0
+readonly EXTENSION_CONFIG=extensionconfig/public.json
+readonly EXTENSION_CONFIG_PROT=extensionconfig/protected.json
+expected_extension_version=
+
+
+### Functions
+
+log() {
+	echo "[$(date +%T)][DEBUG]" "$@"
+}
+
+err() {
+	echo >&2 "[$(date +%T)][ERROR]" "$@"
+}
+
+is_empty() {
```

Comment:
I have a philosophical issue with using these functions. They are simple wrappers for builtin shell functionality, but they start you down the slippery slope to using functions to make bash look like some other programming language or environment. The [ -z "$var" ] and [ -f "$var ] syntaxes are native to the language, instantly understood by an experienced shell script programmer, and should not be wrapped in some function. Creating a function for these purposes forces a human to wonder "gee, I wonder what the test is doing other than simply checking for emptiness of a string, or existence of a file"; it adds a level of abstraction that actually complicates the life of a reviewer.